widget/table.go

```diff
@@ -151,6 +151,101 @@ func (t *Table) UnselectAll() {
 	}
 }
 
+// ScrollTo will scroll to the given cell without changing the selection.
+// Attempting to scroll beyond the limits of the table will scroll to
+// the edge of the table instead.
+//
+// Since: 2.1
+func (t *Table) ScrollTo(id TableCellID) {
+	if t.Length == nil {
+		return
+	}
+
+	rows, cols := t.Length()
+	if id.Row >= rows {
+		id.Row = rows - 1
+	}
+
+	if id.Col >= cols {
+		id.Col = cols - 1
+	}
+
+	t.scrollTo(id)
+}
+
+// ScrollToBottom scrolls to the last row in the table
+//
+// Since: 2.1
+func (t *Table) ScrollToBottom() {
+	if t.Length == nil {
+		return
+	}
+
+	rows, _ := t.Length()
+	t.scrollTo(TableCellID{Row: rows - 1})
+}
+
+// ScrollToLeading scrolls horizontally to the leading edge of the table
+//
+// Since: 2.1
+func (t *Table) ScrollToLeading() {
+	if t.scroll == nil {
+		return
+	}
+
+	t.scroll.Offset.X = 0
+	t.offset.X = 0
+	if t.moveCallback != nil {
+		t.moveCallback()
+	}
+	t.scroll.Refresh()
+	t.cells.Refresh()
+}
+
+// ScrollToTop scrolls to the first row in the table
+//
+// Since: 2.1
+func (t *Table) ScrollToTop() {
```

Comment:
This will also scroll to the first column, is that intended?

Reply:
It was, based on your previous comment about scrolling to the beginning of the last row being more useful than to the end of the last row.

However, with the addition of ScrollToLeading (and Trailing) then I guess this should just scroll vertically.

Reply:
Apologies for the implication, I had not meant that we should reset the X axis value.